recommender/static/js/src/recommender.js

```diff
@@ -1,1022 +1,1187 @@
+     */
+    function resetImportResourcePage() {
+        $('.importResourceFile', element).val('');
+        $('.importResourceSubmit', element).attr('disabled', true);
+    }
+
+    /**
+     * Hide resource list and show pages for modifying
+     * (i.e., add/edit/endorse/remove/import/flag) recommender
+     * @param {string} page The string indicating the page we are going to show. 
+     */
+    function showModifyingPage(page) {
+        $(page, element).show().attr('aria-hidden', 'false');
+        $(page, element).find('.modifyPageTitle').text(modifyPageTitle[page]);
+        $('.recommenderContent', element).hide().attr('aria-hidden', 'true');
+        $('.recommenderModify', element).show().attr('aria-hidden', 'false').attr('aria-label', headerText[page]).focus();
+        $('.recommenderModifyTitle', element).text(headerText[page]);
+    }
+
+    /**
+     * Import resources into the recommender.
+     */
+    function bindImportResourceEvent() {
+        Logger.log('mit.recommender.importResource', generateLog(loggerStatus['importResource']['attempt']));
+
+        resetImportResourcePage();
+        showModifyingPage('.importResourcePage');
+        $('.importResourceFile', element).change(function() { $('.importResourceSubmit', element).attr('disabled', false); });
+
+        $('.importResourceSubmit', element).click(function() {
+            var formDiv = $('.importResourceForm', element);
+            var file = new FormData($(formDiv)[0]);
+
+            $.ajax({
+                type: 'POST',
+                url: importResourceUrl,
+                data: file,
+                contentType: false,
+                cache: false,
+                processData: false,
+                async: false,
+                success: function(result) {
+                    /* Rendering new resources */
+                    $('.recommenderResource', element).remove();
+                    for (var resource_id in result['recommendations']) {
+                        item = result['recommendations'][resource_id];
+                        var newResourceDiv = showResourceEntry(item['upvotes'] - item['downvotes'], item);
+
+                        if (result['endorsed_recommendation_ids'].indexOf(resource_id) !== -1){
+                            $('.endorse', newResourceDiv).addClass('endorsed');
+                            $('.recommenderEndorseReason', newResourceDiv).text(result['endorsed_recommendation_reasons'][result['endorsed_recommendation_ids'].indexOf(resource_id)]);
+                        }
+                    }
+                    paginationItem();
+                    pagination();
+                    backToView();
+                    Logger.log('mit.recommender.importResource', generateLog(loggerStatus['importResource']['complete'], result));
+                },
+                error: function(result, status) {
+                    alert(importResourceErrorText[result.status]);
```

Comment:
It's annoying that jQuery doesn't automatically parse the response for you in the `error` callback like it does in the `success` callback, but that doesn't mean you can't parse it yourself. You can do:
```js
error: function(xhr, status) {
  var data = JSON.parse(xhr.responseText);
  // if the server included an error message, display it
  if (data.error) {
    alert(data.error);
  }
  // continue processing error response as needed
}
```